### DISCUSSION TITLE
Add ID Segment to CCD

### DISCUSSION THREAD

**Discussion by LLCJK** [nextgenhealthcare/connect#5596] [Q&A]
So in a CCD there is the <patientRole> tag and inside of it I have 2 sets of <id> tags. I am trying to add a 3rd <id> tag after the 2nd one. I loop through the id tags and when I determine I have gotten to the final 1 I try to add a 3rd like this but it is adding it in the middle of the 2nd tag..

i == length-1 would symbolize we have just finished looping through all of the id tags. I want to add a id tag after that

	if(i == msg['recordTarget']['patientRole']['id'].length() - 1){
		msg['recordTarget']['patientRole']['id'] += "<id root='1234'/>"
	}

**Comment by wi4seo** [comment]
Hey buddy,

FREE Registration for Mirth Unconference 2023 is Now Opened!

To know more, please visit the event page and register yourself: https://wi4.org/mirth-connect-unconference-feb-2023/

Stay Tuned for More Updates!​

​Thanks
